src/streaming.ts

```diff
@@ -58,13 +60,21 @@ export class Stream<Item> implements AsyncIterable<Item> {
           }
 
           if (sse.event === null) {
+            let data;
+
             try {
-              yield JSON.parse(sse.data);
+              data = JSON.parse(sse.data);
             } catch (e) {
               console.error(`Could not parse message into JSON:`, sse.data);
               console.error(`From chunk:`, sse.raw);
               throw e;
             }
+
+            if (data && data.error) {
+              throw new APIError(undefined, data.error, undefined, undefined);
```

Comment:
Looks like we're not setting the `APIError.message` here, is that intentional?

Reply:
Yes, I was under the impression that the API error data will include a `message` property. That property will then be used by the `APIError` class if a message isn't explicitly passed in the constructor. If the message property isn't set, then we'll default to stringifying the error object to raw JSON.

Is that assumption correct?

Reply:
It's this line https://github.com/openai/openai-node/blob/648403920c954184585973fcd36e06b3bf36549d/src/error.ts#L36

The `error` argument there is the data that the API responded with

Reply:
Ah I see, thanks!

When status is undefined, should we skip the space in the error message here?

https://github.com/openai/openai-node/blob/648403920c954184585973fcd36e06b3bf36549d/src/error.ts#L22

Reply:
Yeah we can probably clean that up; I'll file a ticket